Resources/Public/JavaScript/ApplicationsModule.js

```diff
@@ -77,6 +77,11 @@ require(
 			});
 		});
 
+        // Destroy the dataTable on submit to keep the right order in the BFCache
+        $('#newMassNotification').submit(function(){
+            $('.applications-list').dataTable().fnDestroy();
```

Comment:
Please use `$(this).find('.applications-list')...` instead here, this avoids unexpected side-effects.